### ISSUE TITLE
History page doesn't show results

### ISSUE THREAD

**Issue by mwasilew** [Linaro/squad#875]
Example:
https://qa-reports.linaro.org/android-lkft/5.4-gki-aosp-master-db845c/build/5.4.72-b34934d40c49/testrun/3332171/suite/cts-lkft$arm64-v8a.CtsLibcoreTestCases/test/com.android.org.conscrypt.CertBlacklistTest.testBlacklistedPublicKey/history/?page=1

The page is passed on the build version 5.4.71-0dd22903aee4:
https://qa-reports.linaro.org/android-lkft/5.4-gki-aosp-master-db845c/build/5.4.71-0dd22903aee4/testrun/3313166/suite/cts-lkft$arm64-v8a.CtsLibcoreTestCases/tests/

**Comment by mwasilew** [comment]
@amrohassaan could you take a look? It's kinda urgent.

**Comment by amrohassaan** [comment]
@mwasilew i can see the build there in history? is this what you meant...that it is missing?

**Comment by mwasilew** [comment]
@amrohassaan _results_ are missing

**Comment by mwasilew** [comment]
Similar issue happens on comparison page:
https://qa-reports.linaro.org/_/comparetest/?project=261&suite=cts-lkft%2Farm64-v8a.CtsLibcoreTestCases&test=com.android.org.conscrypt.CertBlacklistTest.testBlacklistedPublicKey
All results come as 'n/a' which is incorrect as the results are present.

**Comment by amrohassaan** [comment]
@mwasilew after debugging found that the 'suite' slug is parsed and saved in wrong pattern into SuiteMetadata which is causing history to fail. 
I'd guess and say it is around this line in tradefed plugin:
https://github.com/Linaro/squadplugins/blob/107ef69669ea6ac8519b5a812c433289efac06b4/tradefed/__init__.py#L46
For example [this](https://qa-reports.linaro.org/android-lkft/5.4-gki-aosp-master-db845c/build/5.4.71-206b4c285dac/testrun/3329218/suite/cts-lkft$arm64-v8a.CtsOsTestCases/test/android.os.cts.AutoRevokeTest.testAutoRevoke_userWhitelisting/details/
) suite name is **'cts-lkft/arm64-v8a.CtsOsTestCases'** but in SuiteMetadata for this test it is saved as **'arm64-v8a.CtsOsTestCases/android.os.cts.AutoRevokeTest'**

**Comment by amrohassaan** [comment]
debugging tredfed locally, [this line](https://github.com/Linaro/squadplugins/blob/107ef69669ea6ac8519b5a812c433289efac06b4/tradefed/__init__.py#L46) received a wrong `module_name`. prints **arm64-v8a.CtsDeqpTestCases/dEQP-EGL.functional.create_context_ext.gl_30** as suite name instead of **cts[something]/arm64-v8a.CtsDeqpTestCases**.

**Comment by mwasilew** [comment]
Thanks @amrohassaan. However I'm not sure this is the problem. This line is a copy/paste from version 1.15 which used to work: https://github.com/Linaro/squadplugins/blob/1.15/tradefed/__init__.py#L148

**Comment by amrohassaan** [comment]
@mwasilew sure

**Comment by mwasilew** [comment]
@amrohassaan there is a good chance that the plugin didn't work properly after changing to processing of 'aggregated' results. We will need to fix all of that in the database I'm afraid :(

**Comment by amrohassaan** [comment]
@mwasilew yes, was obvious the db got corrupted slightly in the metadata table and this will need cleaning.

**Comment by amrohassaan** [comment]
[here](https://qa-reports.linaro.org/android-lkft/5.4-gki-aosp-master-db845c-presubmit/build/5.4.69-489646b879ed/testrun/3277396/suite/cts-lkft$arm64-v8a.CtsDeqpTestCases/tests/) is a correct working version of the test:
**dEQP-EGL.functional.create_context_ext.gl_30.rgb565_depth_no_stencil** and history working properly with the exception i don't see the env column.

**Comment by mwasilew** [comment]
@amrohassaan I don't understand. The missing column is exactly what the problem is. Could you rephrase so it's clear what you mean?

**Comment by amrohassaan** [comment]
@mwasilew in this line: https://github.com/Linaro/squadplugins/blob/107ef69669ea6ac8519b5a812c433289efac06b4/tradefed/__init__.py#L223
**'atomic_test_suite_name'** is composed and suite and suitemetadata for this suite is created, but then we pass **'module_name'** to create the testcase in this line:
https://github.com/Linaro/squadplugins/blob/107ef69669ea6ac8519b5a812c433289efac06b4/tradefed/__init__.py#L231
and then reconstruct the atomic_suite name again in this line:
https://github.com/Linaro/squadplugins/blob/107ef69669ea6ac8519b5a812c433289efac06b4/tradefed/__init__.py#L46
So from the tradefed xml file:
**module_name = 'FastbootVerifyUserspaceTest'** if there is no abi attrib OR **'arm64-v8a.FastbootVerifyUserspaceTest'** if there is abi attrib. And then we construct the atomic_test_suite_name again using the **module_name** and now it becomes:
**arm64-v8a.FastbootVerifyUserspaceTest/com.android.tests.fastboot.FastbootVerifyUserspaceTest** and that is saved in the db as the suite for the test Suitemetadata which breaks everything (history and the missing env column in #876 ..both have the same root cause from what i have seen)
I passed the already constructed **'atomic_test_suite_name'** in line 223 to **create_testcase_tests** in line 231 instead of the module_name and test suitemetadata gets the correct suite name as **"vts-test/arm64-v8a.FastbootVerifyUserspaceTest"**
I fetched a bunch of android cts and vts jobs locally and history works fine now and environements show properly.
if you can pull this https://github.com/amrohassaan/squadplugins/tree/fix_tradefed_metadata_parsing_bug
and this https://github.com/amrohassaan/squad/tree/fix_suite_parsing_for_test_details_view and fetch an android job locally and confirm it is working or they are wrong, that would be great.
If this fix is proper then Monday @chaws and myself will work on fixing the corrupt suite col for the affected tests in SuiteMetadata.
